model/Image.php

```diff
@@ -7,8 +7,8 @@
  * @subpackage filesystem
  */
 class Image extends File {
-	public function __construct($record = null, $isSingleton = false, $model = null) {
-		parent::__construct($record, $isSingleton, $model);
+	public function __construct($record = null, $isSingleton = false, $model = null, $queryParams = array()) {
+		parent::__construct($record, $isSingleton, $model, $queryParams);
```

Comment:
Right, so this shows the problem with mucking with the constructor. Every class that overrides it now needs to do this or sad times in sad town for sad people.